### PR TITLE
docs: Azure DevOps Pipelines

### PR DESCRIPTION
**Pull request by devcatalin**
## Pull request description 

Added documentation on how to use the Azure DevOps Extension to integrate with Testkube

## Checklist (choose whats happened)

- [ ] breaking change! (describe)
- [ ] tested locally
- [ ] tested on cluster
- [ ] added new dependencies
- [x] updated the docs
- [ ] added a test

## Breaking changes

-

## Changes

-

## Fixes

-

### CODE REVIEW

File: docs/docs/articles/azure.md [hunk 1]
```diff
@@ -0,0 +1,159 @@
+# Testkube Azure DevOps Pipelines
+
+The Testkube Azure DevOps integration streamlines the installation of Testkube, enabling the execution of any [Testkube CLI](https://docs.testkube.io/cli/testkube) command within Azure DevOps pipelines. This integration can be effortlessly integrated into your Azure DevOps setup, enhancing your continuous integration and delivery processes.
```
Comment (jfermi) [inline docs/docs/articles/azure.md:3]:
```suggestion
Testkube's integration with Azure DevOps streamlines the installation of Testkube, enabling the execution of any [Testkube CLI](https://docs.testkube.io/cli/testkube) command within Azure DevOps pipelines. This integration can be effortlessly integrated into your Azure DevOps setup, enhancing your continuous integration and delivery processes.
```

File: docs/docs/articles/azure.md [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+This Azure DevOps integration offers a versatile solution for managing your pipeline workflows and is compatible with Testkube Pro, Testkube Enterprise, and the open-source Testkube platform. It allows Azure DevOps users to effectively utilize Testkube's capabilities within their CI/CD pipelines, providing a robust and flexible framework for test execution and automation.
```
Comment (jfermi) [inline docs/docs/articles/azure.md:4]:
```suggestion
The Azure DevOps integration offers a versatile solution for managing your pipeline workflows and is compatible with Testkube Pro, Testkube Enterprise, and the open-source Testkube platform. It allows Azure DevOps users to effectively utilize Testkube's capabilities within their CI/CD pipelines, providing a robust and flexible framework for test execution and automation.
```

File: docs/docs/articles/azure.md [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+
+### Azure DevOps Extension
+
+Install the Testkube CLI extension using the following url:
+[https://marketplace.visualstudio.com/items?itemName=Testkube.testkubecli](https://marketplace.visualstudio.com/items?itemName=Testkube.testkubecli)
+
+## Testkube Pro
+
+### How to configure Testkube CLI action for Testkube Pro and run a test
```
Comment (jfermi) [inline docs/docs/articles/azure.md:13]:
```suggestion
### How to configure Testkube CLI action for Testkube Pro and run a test.
```

File: docs/docs/articles/azure.md [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+          organization: '$(TK_ORG_ID)'
+          environment: '$(TK_ENV_ID)'
+          token: '$(TK_API_TOKEN)'
+      - script: testkube run test test-name -f
+        displayName: Run Testkube Test
+```
+
+## Testkube OSS
+
+### How to configure Testkube CLI action for TK OSS and run a test
```
Comment (jfermi) [inline docs/docs/articles/azure.md:45]:
```suggestion
### How to configure the Testkube CLI action for TK OSS and run a test.
```

File: docs/docs/articles/azure.md [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+        inputs:
+          namespace: 'custom-testkube-namespace'
+          url: 'custom-testkube-url'
+      - script: testkube run test test-name -f
+        displayName: 'Run Testkube Test'
+```
+
+The steps to connect to your Kubernetes cluster differ for each provider. You should check the docs of your Cloud provider for how to connect to the Kubernetes cluster from Azure DevOps Pipelines
+
+### How to configure Testkube CLI action for TK OSS and run a test
```
Comment (jfermi) [inline docs/docs/articles/azure.md:73]:
```suggestion
### How to configure Testkube CLI action for TK OSS and run a test.
```

File: docs/docs/articles/azure.md [hunk 6]
```diff
@@ -0,0 +1,159 @@
...
+
+This workflow establishes a connection to EKS cluster and creates and runs a test using TK CLI. In this example we also use variables not
```
Comment (jfermi) [inline docs/docs/articles/azure.md:75]:
```suggestion
This workflow establishes a connection to the EKS cluster and creates and runs a test using TK CLI. In this example we also use variables not
```

File: docs/docs/articles/azure.md [hunk 7]
```diff
@@ -0,0 +1,159 @@
...
+ to reveal sensitive data. Please make sure that the following points are satisfied:
+- The **AWS_ACCESS_KEY_ID**, **AWS_SECRET_ACCESS_KEY** secrets should contain your AWS IAM keys with proper permissions to connect to EKS cluste
```
Comment (jfermi) [inline docs/docs/articles/azure.md:77]:
```suggestion
- The **AWS_ACCESS_KEY_ID**, **AWS_SECRET_ACCESS_KEY** secrets should contain your AWS IAM keys with proper permissions to connect to the EKS cluster.
```

File: docs/docs/articles/azure.md [hunk 8]
```diff
@@ -0,0 +1,159 @@
...
+r.
```
Comment (jfermi) [inline docs/docs/articles/azure.md:78]:
```suggestion
```

File: docs/docs/articles/azure.md [hunk 9]
```diff
@@ -0,0 +1,159 @@
...
+- The **AWS_REGION** secret should contain AWS region where EKS is
+- Tke **EKS_CLUSTER_NAME** secret points to the name of EKS cluster you want to connect.
```
Comment (jfermi) [inline docs/docs/articles/azure.md:80]:
```suggestion
- Tke **EKS_CLUSTER_NAME** secret points to the name of the EKS cluster you want to connect.
```

File: docs/docs/articles/azure.md [hunk 10]
```diff
@@ -0,0 +1,159 @@
...
+          organization: '$(TK_ORG_ID)'
+          environment: '$(TK_ENV_ID)'
+          token: '$(TK_API_TOKEN)'
+
+      - script: testkube run test test-name -f
+        displayName: Run Testkube Test
+
+```
+
+### How to connect to GKE (Google Kubernetes Engine) cluster and run a test
```
Comment (jfermi) [inline docs/docs/articles/azure.md:115]:
```suggestion
### How to connect to GKE (Google Kubernetes Engine) cluster and run a test.
```

File: docs/docs/articles/azure.md [hunk 11]
```diff
@@ -0,0 +1,159 @@
...
+          token: '$(TK_API_TOKEN)'
+
+      - script: testkube run test test-name -f
+        displayName: Run Testkube Test
+
+```
+
+### How to connect to GKE (Google Kubernetes Engine) cluster and run a test 
+
+This example connects to a k8s cluster in Google Cloud, creates and runs a test using Testkube Azure DevOps pipeline. Please make sure that the following points are satisfied:
```
Comment (jfermi) [inline docs/docs/articles/azure.md:117]:
```suggestion
This example connects to a k8s cluster in Google Cloud and creates and runs a test using Testkube Azure DevOps pipeline. Please make sure that the following points are satisfied:
```

File: docs/docs/articles/azure.md (hunk 10 above)
Comment (jfermi) [inline docs/docs/articles/azure.md:115]:
```suggestion
### How to connect to GKE (Google Kubernetes Engine) cluster and run a test.
```

File: docs/docs/articles/azure.md [hunk 12]
```diff
@@ -0,0 +1,159 @@
...
+- The **GKE Sevice Account** should be created prior in Google Cloud and added to pipeline variables along with **GKE_PROJECT** value;
```
Comment (jfermi) [inline docs/docs/articles/azure.md:118]:
```suggestion
- The **GKE Sevice Account** should have already been created in Google Cloud and added to pipeline variables along with **GKE_PROJECT** value.
```

File: docs/docs/articles/jenkins-ui.md [hunk 13]
```diff
@@ -0,0 +1,33 @@
...
+[https://plugins.jenkins.io/testkube-cli](https://plugins.jenkins.io/testkube-cli)
+
+## Testkube Pro
+
+To use Jenkins CI/CD for [Testkube Pro](https://app.testkube.io/), you need to create an [API token](https://docs.testkube.io/testkube-pro/articles/organization-management/#api-tokens).
+
+
+### How to set up a Freestyle Project to run tests on Testkube Pro
+
+1. Create a new Freestyle Project
```
Comment (jfermi) [inline docs/docs/articles/jenkins-ui.md:19]:
```suggestion
1. Create a new Freestyle Project.
```

File: docs/docs/articles/jenkins-ui.md [hunk 14]
```diff
@@ -0,0 +1,33 @@
...
+2. In General settings, configure the environment variables:
+   - TK_ORG
+   - TK_ENV
+   - TK_API_TOKEN
+  
+![jenkins environment variables configuration](../img/jenkins-environment.png)
+
+3. Click on "Add Build Step" and select "Testkube Setup"
```
Comment (jfermi) [inline docs/docs/articles/jenkins-ui.md:27]:
```suggestion
3. Click on "Add Build Step" and select "Testkube Setup".
```

File: docs/docs/articles/jenkins-ui.md [hunk 15]
```diff
@@ -0,0 +1,33 @@
...
+![jenkins testkube setup build step](../img/jenkins-build-step.png)
+
+4. Specify a Testkube CLI version or leave it empty to use the latest version
+
+5. Add a new "Execute Shell" Build Step and run one or multiple Testkube CLI commands
```
Comment (jfermi) [inline docs/docs/articles/jenkins-ui.md:32]:
```suggestion
5. Add a new "Execute Shell" Build Step and run one or multiple Testkube CLI commands.
```

**Review by jfermi — COMMENTED** [review]
Some editing suggestions.

**Review by vsukhin — APPROVED** [review]

**Pull request merged**